### DISCUSSION TITLE
Unable to get recursive groups

### DISCUSSION THREAD

**Discussion by huggla** [DirectoryTree/LdapRecord#258] [Q&A]
I'm trying to get all groups of a user. Running the code below takes several seconds and ends with a php error 500. It works when I remove recursive(), but then I don't get all groups. Both adldap and adldap2 handles recursive groups fine.

```
use LdapRecord\Container;
use LdapRecord\Connection;
use LdapRecord\Models\ActiveDirectory\User;

$connection = new Connection([...]);
Container::addConnection($connection);
$user = User::find('CN=HEUG001,OU=2282,OU=Employees,OU=ADM,DC=kristianstad,DC=se');
$groups = $user->groups()->recursive()->get();
```

**Comment by stevebauman** [comment]
Hi @huggla,

> takes several seconds and ends with a php error 500.

Can you post the stack trace / error logs associated with the 500 error?

**Reply by huggla** [reply]
Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /wkarta-internt/php/functions/vendor/directorytree/ldaprecord/src/Models/Concerns/HasAttributes.php on line 870
